File: reservation-frontend/src/app/core/models/service/service.ts
import { Salle } from '../salle/salle';
import { Reservation } from '../reservations/reservation';

export class Service {
  serviceId?: number;
  nom!: string;
  description?: string;
  categorie? : string;
  statut? : string;
  prix!: number;
  salleId?: number;
  reservations?: Reservation[] = [];
  salle?: Salle;

  constructor(
    serviceId?: number,
    nom?: string,
    prix?: number,
    description?: string,
    categorie?: string,
    statut?: string,
    salleId?: number,
    reservations: Reservation[] = [],
    salle?: Salle
  ) {
    if (serviceId !== undefined) this.serviceId = serviceId;
    if (nom) this.nom = nom;
    if (prix !== undefined) this.prix = prix;
    if (description) this.description = description;
    if (salleId) this.salleId = salleId;
    if (categorie) this.categorie = categorie;
    if (statut) this.statut = statut;
    if (salle) this.salle = salle;

    this.reservations = reservations;
  }
}

File: Reservation/src/main/java/net/travel/reservation/entites/Service.java
package net.travel.reservation.entites;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import lombok.*;

import java.math.BigDecimal;
import java.util.ArrayList;
import java.util.List;

@Entity
@Table(name = "services")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Service {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long serviceId;

    @Column(nullable = false)
    private String nom; // Ex: "DJ", "Décoration", "Traiteur", "Photographe"
    @Column(length = 500)
    private String description;
    private BigDecimal prix; // Prix du service en EUR/TND...
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "salle_id", nullable = false)
    private Salle salle;
    @ManyToMany(mappedBy = "services")
    @JsonIgnore // Eviter la boucle de référence JSON
    @Builder.Default
    private List<Reservation> reservations = new ArrayList<>();
}